core/Plugin/Manager.php

```diff
@@ -1654,5 +1662,47 @@ public function loadPluginTranslations()
             $translator->addDirectory(self::getPluginDirectory($pluginName) . '/lang');
         }
     }
+
+    /**
+     * @param string $pluginName
+     * @return Date|null
+     * @throws \Exception
+     */
+    public function getPluginLastActivationTime($pluginName)
```

Comment:
not any important but could move this methods to the `Plugin` class so things would maybe feel a bit more natural like `$plugin->getLastActivationTime()`. Not sure that works though in our context maybe we need to have this information when the plugin is not loaded yet or unloaded already or so. Feel free to ignore